src/Kros.AspNetCore/Extensions/HttpClientFactoryExtensions.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Microsoft.AspNetCore.Http;
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+
+namespace Kros.AspNetCore.Net
+{
+    /// <summary>
+    /// Extensions for HttpClientFactory.
+    /// </summary>
+    public static class HttpClientFactoryExtensions
+    {
+        /// <summary>
+        /// Creates HttpClient with authorization header from <paramref name="httpContextAccessor"/>.
+        /// </summary>
+        public static HttpClient CreateClientWithAuthorization(this IHttpClientFactory httpClientFactory, IHttpContextAccessor httpContextAccessor)
+        {
+            HttpClient client = httpClientFactory.CreateClient();
+            try
+            {
+                string authHeader = httpContextAccessor.HttpContext.Request.Headers["Authorization"].FirstOrDefault();
+                if (authHeader != null)
+                {
+                    AuthenticationHeaderValue.TryParse(authHeader, out AuthenticationHeaderValue newVal);
+                    client.DefaultRequestHeaders.Authorization = newVal;
+                }
+            }
+            catch (Exception)
```

Comment:
You can simplified it.
```suggestion
            catch
```